Font.js

```diff
@@ -115,7 +115,8 @@ class Font extends EventManager {
      *
      * @param {Buffer} buffer The binary data associated with this font.
      */
-    async fromDataBuffer(buffer, type) {
+    async fromDataBuffer(buffer, filenameOrUrl) {
+        const type = getFontCSSFormat(filenameOrUrl);
```

Comment:
```suggestion
    async fromDataBuffer(buffer, typeOrPath) {
        let type = typeOrPath;
        if([`ttf`,`otf`,...].indexOf(typeOrPath) === -1) {
            type = getFontCSSFormat(typeOrPath);
        }
```

Reply:
let's make that a combination of two lists at the top, too: e.g.
```
const PERMITTED_TYPES = [
    `ttf`,
    `otf``,
    `woff`,
    `woff2`,
];
const ILLEGAL_TYPES = [
    `eot`,
    `svg`,
    `fon`,
    `ttc`.
];
consts ALL_TYPES = [...PERMITTED_TYPES, ...ILLEGAL_TYPES]
```

and then we can say 

```
if (!ALL_TYPES.includes(typeOrPath)) {
  ...
}
```

Reply:
Good idea! Implemented! :)